src/hotspot/share/gc/shared/partialArrayTaskStats.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "gc/shared/partialArrayTaskStats.hpp"
+#include "logging/log.hpp"
+#include "logging/logHandle.hpp"
+#include "logging/logStream.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/growableArray.hpp"
+#include "utilities/macros.hpp"
+#include "utilities/ostream.hpp"
+
+#if TASKQUEUE_STATS
+
+PartialArrayTaskStats::PartialArrayTaskStats()
+  : _split(0), _pushed(0), _stolen(0), _processed(0)
+{}
+
+void PartialArrayTaskStats::accumulate(const PartialArrayTaskStats& stats) {
+  _split += stats._split;
+  _pushed += stats._pushed;
+  _stolen += stats._stolen;
+  _processed += stats._processed;
+}
+
+void PartialArrayTaskStats::reset() {
+  *this = PartialArrayTaskStats();
```

Comment:
Can we do sth like `static_assert(std::is_trivially_copyable<PartialArrayTaskStats>::value)` here?

Reply:
I think you mean is_trivially_assignable.  I don't think it's a useful
assertion here.  Depending on details of the class, one might reasonably
implement such an operation in the same way even if it isn't trivially
assignable.